### PR TITLE
Iracing replay keymap

### PR DESCRIPTION
**Pull request by AaronVerDow**
## Description

Adds a keymap for an iRacing replay keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+void tap_key(uint16_t keycode) {
+  register_code  (keycode);
+  unregister_code(keycode);
+}
+
+void shift_key(uint16_t keycode) {
+  register_code  (KC_LSFT);
+  tap_key        (keycode);
+  unregister_code(KC_LSFT);
+}
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:12]: These functions already exist as `tap_code` and `tap_code16`.

Though, for `tap_code16`, you'd want `tap_code16(LSFT(kc))`

```suggestion
```

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    REVERSE
+};
+
+void cam_up(qk_tap_dance_state_t *state, void *user_data) {
+    switch (state->count) {
+        case 1:
+            tap_key(KC_C);           // next cam
+            break;
+        case 2:
+            shift_key(KC_8);
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:40]: https://docs.qmk.fm/#/keycodes_us_ansi_shifted?id=keycodes

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:177]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c (hunk 3 above)
Comment (AaronVerDow) [reply keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c]: This change looks straight forward enough but it doesn't compile:

## Diff
```
diff --git a/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c b/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c
index 983d73a3f..0e99eb68e 100644
--- a/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c
+++ b/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c
@@ -46,9 +46,9 @@ void cam_up(qk_tap_dance_state_t *state, void *user_data) {
             tap_code(KC_C);           // next cam
             break;
         case 2:
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
             tap_code(KC_2);
             wait_ms(CAM_DELAY);
@@ -65,9 +65,9 @@ void cam_down(qk_tap_dance_state_t *state, void *user_data) {
             tap_code16(LSFT(KC_C));           // next cam
             break;
         case 2:
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
             tap_code(KC_2);
             wait_ms(CAM_DELAY);
```
## Error
```
Ψ Compiling keymap with make -j 1 kbdfans/kbdpad_mk2:iracing_replay


QMK Firmware 0.12.29
WARNING: Some git submodules are out of date or modified.
 Please consider running make git-submodule.

Making kbdfans/kbdpad_mk2 with keymap iracing_replay

avr-gcc (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text	  data	   bss	   dec	   hex	filename
      0	 26564	     0	 26564	  67c4	.build/kbdfans_kbdpad_mk2_iracing_replay.hex

Compiling: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c                            In file included from quantum/keymap.h:47,
                 from quantum/quantum.h:30,
                 from keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h:18,
                 from keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:17:
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c: In function 'cam_up':
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:49:22: note: in expansion of macro 'KC_ASTR'
   49 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:51:22: note: in expansion of macro 'KC_ASTR'
   51 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c: In function 'cam_down':
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:68:22: note: in expansion of macro 'KC_ASTR'
   68 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:70:22: note: in expansion of macro 'KC_ASTR'
   70 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
cc1: all warnings being treated as errors
 [ERRORS]
 | 
 | 
 | 
make[1]: *** [tmk_core/rules.mk:417: .build/obj_kbdfans_kbdpad_mk2_iracing_replay/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.o] Error 1
Make finished with errors
make: *** [Makefile:523: kbdfans/kbdpad_mk2:iracing_replay] Error 1
```

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c (hunk 1 above)
Comment (AaronVerDow) [reply keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c]: Added.

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c (hunk 3 above)
Comment (AaronVerDow) [reply keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c]: Nevermind, figured it out.  Needed to stick with `tap_code16()`.

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c (hunk 4 above)
Comment (AaronVerDow) [reply keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c]: Removed.

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c (hunk 2 above)
Comment (AaronVerDow) [reply keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c]: Glad to find those.  Fixed.

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/readme.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# iRacing Replay Tool
+
+![](https://imgur.com/0948eBG.jpg)
```
Comment (fauxpark) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/readme.md:3]:
```suggestion
![](https://imgur.com/0948eBGh.jpg)
```

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+  switch(keycode) {
+        // these are not used currently, leaving them in case someone wants to bind them to a key instead of tap dance
+        case CHASE:
+            if (record->event.pressed) {
+               SEND_STRING("**20" SS_TAP(X_ENTER));
+            }
+            break;
+        case REVERSE:
+            if (record->event.pressed) {
+               SEND_STRING("**22" SS_TAP(X_ENTER));
```
Comment (fauxpark) [inline keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:176]:
```suggestion
               SEND_STRING("**20\n");
            }
            break;
        case REVERSE:
            if (record->event.pressed) {
               SEND_STRING("**22\n");
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c (hunk 6 above)
Comment (AaronVerDow) [reply keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c]: Did some more thorough testing and found the game won't recognize this combo so I removed it.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**